### DISCUSSION TITLE
Exporting .syx files for DX 7

### DISCUSSION THREAD

**Discussion by sylvesp** [christofmuc/KnobKraft-orm#303] [Q&A]
First of all... sorry for my ignorance, I am a newbie to all this - but will catch up fast :) Promise!

I can import into KnobKraft  .syx files that contain DX 7 patches (actually used in [dexed](https://asb2m10.github.io/dexed/) )
I can sort/arrange the patches in KnobKraft 
=> I would like to export a selection of patches into a .syx file that Dexed can load - like the originals from where the import did happen.

No matter what I did try, the exported .syx files cannot be loaded into dexed. There is an error message that the file doesn't have a valid format.

(After installing a LoopMIDI I can even send patches loaded into KnobKraft  directly into my dexed plugin running in Studio One on the same PC! - Yay!)

Quesion: How can I export .syx files from KnobKraft that are compatible with the DX 7  patch/cartridge files?
Any suggestions about what am I missing or doing wrong?
Appreciate your help!

**Comment by christofmuc** [comment]
Yikes, very good question here! I need to dig in and see what's missing. It might be actually that it is not implemented (as in #296), or it might be that there are some subtle bits missing in the export. I'll check!

**Comment by christofmuc** [comment]
I looked at the Dexed source code, and it seems it can only import DX7 bank dump files, not the DX7 single voice format that KnobKraft uses to be able to send patches one by one.

So there are two issues preventing this to work right now:

#296 which would allow the UI to allow a full bank to be saved to disk in bank format

#306 which would allow the adaptation to expose a function that can generate a full bank dump.

Sorry no better answer right now, but I took both required items on the to do list! Thanks for the feedback!
